### PR TITLE
save timepoints after connect and not in contructor, adjusted tests

### PR DESCRIPTION
**Pull request by Kumamon38**
I think this should fix the random error of Issue #70

### CODE REVIEW

File: ixwebsocket/IXWebSocketTransport.cpp [hunk 1]
```diff
@@ -234,6 +231,23 @@ namespace ix
...
+            _lastSendPingTimePoint = std::chrono::steady_clock::now();
+        } 
+        {
+            std::lock_guard<std::mutex> lock(_lastReceivePongTimePointMutex);
+            _lastReceivePongTimePoint = std::chrono::steady_clock::now();
+        }
+
+        if (_pingIntervalOrTimeoutGCDSecs > 0)
+        {
+            _nextGCDTimePoint = std::chrono::steady_clock::now() + std::chrono::seconds(_pingIntervalOrTimeoutGCDSecs);
```
Comment (Dimon4eg) [inline ixwebsocket/IXWebSocketTransport.cpp]: The same calculation is in `configure` method. Better to move it to new method to avoid duplication of code

Comment (Kumamon38) [reply ixwebsocket/IXWebSocketTransport.cpp]: I already removed it from the configure method

File: ixwebsocket/IXWebSocketTransport.cpp [hunk 2]
```diff
@@ -173,6 +168,7 @@ namespace ix
                                                          timeoutSecs);
         if (result.success)
         {
+            initTimePointsAndGCDAfterConnect();
```
[the remark this answers is not in the document]
Comment (Kumamon38) [reply ixwebsocket/IXWebSocketTransport.cpp]: yep

File: ixwebsocket/IXWebSocketTransport.cpp [hunk 3]
```diff
@@ -201,6 +197,7 @@ namespace ix
         auto result = webSocketHandshake.serverHandshake(fd, timeoutSecs);
         if (result.success)
         {
+            initTimePointsAndGCDAfterConnect();
```
[the remark this answers is not in the document]
Comment (Kumamon38) [reply ixwebsocket/IXWebSocketTransport.cpp]: yep

File: ixwebsocket/IXWebSocketTransport.cpp (hunk 1 above)
Comment (Dimon4eg) [reply ixwebsocket/IXWebSocketTransport.cpp]: okay

**Comment by bsergean** [comment]
Ah it broke again after merging (on macOS)

https://travis-ci.com/machinezone/IXWebSocket/jobs/199740940

It is still unreliable.

**Pull request merged**